lib/model/rofolder.go

```diff
@@ -8,151 +8,130 @@ package model
 
 import (
 	"fmt"
-	"math/rand"
 	"time"
 
 	"github.com/syncthing/syncthing/lib/sync"
 )
 
 type roFolder struct {
 	stateTracker
+	scan folderscan
 
-	folder    string
-	intv      time.Duration
-	timer     *time.Timer
-	model     *Model
-	stop      chan struct{}
-	scanNow   chan rescanRequest
-	delayScan chan time.Duration
+	folderID string
+	model    *Model
+	stop     chan struct{}
 }
 
-type rescanRequest struct {
-	subs []string
-	err  chan error
-}
-
-func newROFolder(model *Model, folder string, interval time.Duration) *roFolder {
+func newROFolder(model *Model, folderID string, scanInterval time.Duration) *roFolder {
 	return &roFolder{
 		stateTracker: stateTracker{
-			folder: folder,
-			mut:    sync.NewMutex(),
+			folderID: folderID,
+			mut:      sync.NewMutex(),
+		},
+		scan: folderscan{
+			interval: scanInterval,
+			timer:    time.NewTimer(time.Millisecond),
+			now:      make(chan rescanRequest),
+			delay:    make(chan time.Duration),
 		},
-		folder:    folder,
-		intv:      interval,
-		timer:     time.NewTimer(time.Millisecond),
-		model:     model,
-		stop:      make(chan struct{}),
-		scanNow:   make(chan rescanRequest),
-		delayScan: make(chan time.Duration),
+		folderID: folderID,
```

Comment:
sry, unclear from the diff snippet.

```go
stateTracker: stateTracker{
				folderID: cfg.ID,
				mut:      sync.NewMutex(),
			},
folderID : cfg.ID,
```

I expect to remove `folderID` and nothing changes. The later one shadows the inherited from stateTracker. But the tests are failing. Sth I dont understand yet.